### ISSUE TITLE
missing-param-doc W9015 false positive

### ISSUE THREAD

**Issue by xqgex** [pylint-dev/pylint#8628]
### Bug description

When the variable type is a [union](https://docs.python.org/3/library/typing.html#typing.Union) of [Enum](https://docs.python.org/3/library/enum.html) and string I get a missing parameter documentation error.

### `test.py`

```Python
""" Description of the module. """

from enum import Enum
from typing import Union


class SomeEnum(Enum):
    """ Enum description. """
    MEMBER_1 = 'value 1'
    MEMBER_2 = 'value 2'


class SomeClass:
    """ Class description. """

    def __init__(self) -> None:
        self.name: str = 'lorem ipsum'

    def too_few_public_methods(self) -> str:
        """ Function description.

        :return: A description of the return value.
        :rtype: str
        """
        return self.name

    def test_function(self, name: Union[str, SomeEnum]) -> bool:
        """ Function description.

        :param Union[str, SomeEnum] name: The description of the input variable.
        :return: A description of the return value.
        :rtype: bool
        """
        return name == self.name
```

### Command used

```shell
python3.10 -m pylint --load-plugins pylint.extensions.docparams --accept-no-param-doc false ./test.py
```

### Pylint output

```shell
************* Module test.test
test.py:27:4: W9015: "name" missing in parameter documentation (missing-param-doc)

------------------------------------------------------------------
Your code has been rated at 9.17/10 (previous run: 9.17/10, +0.00)
```

### Expected behavior

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.23/10, +0.77)

### Pylint version

```shell
pylint 2.15.7
astroid 2.12.13
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04 TLS

**Comment by xqgex** [comment]
Note, replacing the `Enum` with a base type like `int` produce the same error message
